net/gateway/netaddress.go

```diff
@@ -0,0 +1,166 @@
+package gateway
+
+import (
+	"errors"
+	"net"
+	"strconv"
+	"strings"
+)
+
+// MaxEncodedNetAddressLength is the maximum length of a NetAddress encoded
+// with the encode package. 266 was chosen because the maximum length for the
+// hostname is 254 + 1 for the separating colon + 5 for the port + 8 byte
+// string length prefix.
+const MaxEncodedNetAddressLength = 266
+
+// A NetAddress contains the information needed to contact a peer.
+type NetAddress string
```

Comment:
nah we're not reusing any of this code lol. Just use `string` for now, and we'll deal with the `IsLocal`/`IsLoopback`/etc. stuff later when we implement that actual gateway.